### PR TITLE
Workaround for Electron issue with setImmediate

### PR DESCRIPTION
**Pull request by sergiou87**
## Description

This PR includes a workaround for https://github.com/electron/electron/issues/29261 that basically replaces calls to `setImmediate` with `setTimeout(…, 0)`.

Given it will need some time in testing, I created a very original abstraction with `setAlmostImmediate` and wrote an eslint rule to make sure we don't use `setImmediate` (@shiftkey I'd love if you could take a look at it and share any feedback… but only if you have time!!!)

After testing it, I haven't noticed any issues, and of course I wasn't able to repro the aforementioned issue after bumping Electron to one of the affected versions, so we could say this temporary workaround is actually useful 😌 

## Release notes

Notes: no-notes

### CODE REVIEW

File: eslint-rules/set-almost-immediate.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      nodeJSImmediateForbidden: `NodeJS.Immediate cannot be used, use AlmostImmediate instead`,
+    },
+    fixable: 'code',
+    schema: [],
+  },
+  create: function (context) {
+    const sourceCode = context.getSourceCode()
+
+    /**
+     * Check each member of the interface body and ensure it is marked `readonly`.
```
Comment (tidy-dev) [inline eslint-rules/set-almost-immediate.js]: Copy/Pasta comment

File: eslint-rules/set-almost-immediate.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
+// @ts-check
+
+/**
+ * set-almost-immediate
```
Comment (tidy-dev) [inline eslint-rules/set-almost-immediate.js:4]: These custom rules are neat!

**Review by tidy-dev — CHANGES_REQUESTED** [review]
✨  Seems like a reasonable work around. I like the name. 😸  

One comment need updated from copy/paste of previous lint rule.

**Comment by shiftkey** [comment]
@sergiou87 ESLint rules look fine to me :shipit:

File: eslint-rules/set-almost-immediate.js (hunk 1 above)
Comment (sergiou87) [reply eslint-rules/set-almost-immediate.js]: Ouch! Fixed now, thanks 😄

**Review by tidy-dev — APPROVED** [review]
✨

**Pull request merged**